### PR TITLE
appliance os security audit

### PR DESCRIPTION
**Pull request by maplain**
Fixes: https://github.com/vmware/vic/issues/4441

### CODE REVIEW

**Comment by maplain** [comment]
no, it returns with 0 unless it runs into some trouble. I'll paste the output here

**Comment by maplain** [comment]
xxx/github.com/vmware/vic# BRIDGE_NETWORK=DPortGroup TEST_TIMEOUT=15m GOVC_INSECURE=1 TEST_DATASTORE=datastore1 TEST_URL_ARRAY="xxx" 
 TEST_USERNAME=xxx TEST_PASSWORD=xxx  robot tests/manual-test-cases/Group20-Security/20-1-Appliance-Audit.robot
==============================================================================
20-1-Appliance-Audit :: Test 20-01 - Appliance-Audit
==============================================================================

TEST_URL=10.192.98.41

Installing VCH to test server...
Installer completed successfully: VCH-0-8023...

Enalbe SSH on vch...
Appliance Audit
Provision lynis to vch appliance...
.
Begin auditing appliance...
Appliance Audit                                                       | PASS |
------------------------------------------------------------------------------
Gathering logs from the test server VCH-0-8023
^CSecond signal will force exit.
20-1-Appliance-Audit :: Test 20-01 - Appliance-Audit                  | FAIL |
Suite teardown failed:
Execution terminated by signal

1 critical test, 0 passed, 1 failed
1 test total, 0 passed, 1 failed
==============================================================================
Output:  /Users/fangyuanl/development/go/src/github.com/vmware/vic/output.xml
Log:     /Users/fangyuanl/development/go/src/github.com/vmware/vic/log.html
Report:  /Users/fangyuanl/development/go/src/github.com/vmware/vic/report.html

**Comment by maplain** [comment]
root@ [ ~ ]# ls
lynis-report.dat  lynis.log
root@ [ ~ ]# cat lynis-report.dat
# Lynis Report
report_version_major=1
report_version_minor=0
report_datetime_start=2017-04-11 01:21:31
auditor=[Not Specified]
lynis_version=2.4.8
os=Linux
os_name=Linux
os_fullname=Linux 4.4.54-1.ph1-esx
os_version=4.4.54-1.ph1-esx
linux_version=
os_kernel_version=4.4.54
os_kernel_version_full=4.4.54-1.ph1-esx
hostname=no-hostname
test_category=all
test_group=all
plugin_directory=/usr/local/lynis/plugins
lynis_update_available=-1
binary_paths=/usr/bin,/usr/sbin,/usr/local/bin,/usr/libexec
binaries_count=681
vm=1
vmtype=vmware
container=0
systemd=1
plugins_enabled=0
hostid2=6aea128b18cd6eee097ba9e768018ce1079d9b48daaf1e2ac473808161d5e6ab
hostid=655adf22a166b0819e6c6353f5717168dd2528bf
running_service_tool=systemctl
running_service[]=getty@tty1
running_service[]=haveged
running_service[]=sshd
running_service[]=systemd-journald
running_service[]=systemd-resolved
running_service[]=systemd-udevd
running_service[]=vic-init
boot_service_tool=systemctl
boot_service[]=autovt@
boot_service[]=dbus-org
boot_service[]=getty@
boot_service[]=haveged
boot_service[]=iptables
boot_service[]=nat
boot_service[]=permissions
boot_service[]=sshd-keygen
boot_service[]=systemd-networkd-wait-online
boot_service[]=systemd-resolved
boot_service[]=vic-init
uptime_in_seconds=114
uptime_in_days=0
boot_loader=unknown
boot_uefi_booted=0
boot_uefi_booted_secure=0
service_manager=systemd
linux_default_runlevel=3
cpu_pae=1
cpu_nx=1
linux_kernel_release=4.4.54-1.ph1-esx
linux_kernel_version=#1-photon SMP Sat Mar 25 01:37:47 UTC 2017
linux_kernel_type=modular
loaded_kernel_module[]=ata_piix
loaded_kernel_module[]=cdrom
loaded_kernel_module[]=drm
loaded_kernel_module[]=drm_kms_helper
loaded_kernel_module[]=evdev
loaded_kernel_module[]=i2c_core
loaded_kernel_module[]=ipt_MASQUERADE
loaded_kernel_module[]=ipt_REJECT
loaded_kernel_module[]=iptable_filter
loaded_kernel_module[]=iptable_mangle
loaded_kernel_module[]=iptable_nat
loaded_kernel_module[]=ipv6
loaded_kernel_module[]=libata
loaded_kernel_module[]=mousedev
loaded_kernel_module[]=nf_conntrack
loaded_kernel_module[]=nf_conntrack_ipv4
loaded_kernel_module[]=nf_defrag_ipv4
loaded_kernel_module[]=nf_nat
loaded_kernel_module[]=nf_nat_ipv4
loaded_kernel_module[]=nf_nat_masquerade_ipv4
loaded_kernel_module[]=nf_reject_ipv4
loaded_kernel_module[]=processor
loaded_kernel_module[]=psmouse
loaded_kernel_module[]=sr_mod
loaded_kernel_module[]=ttm
loaded_kernel_module[]=vmwgfx
loaded_kernel_module[]=vmxnet3
loaded_kernel_module[]=xt_LOG
loaded_kernel_module[]=xt_addrtype
loaded_kernel_module[]=xt_conntrack
linux_config_file=/boot/config-4.4.54-1.ph1-esx
linux_kernel_io_scheduler[]=noop
exception_event[]=KRNL-5830:2|Can not find any vmlinuz or kernel files in /boot, which is unexpected|
memory_size=2053184
memory_units=kB
auth_group_ids_unique=1
auth_group_names_unique=1
real_user[]=root,0
real_user[]=vicadmin,1000
pam_cracklib=1
suggestion[]=AUTH-9286|Configure minimum password age in /etc/login.defs|-|-|
suggestion[]=AUTH-9328|Some umasks found could be more strict (e.g. 027)|-|-|
manual_event[]=AUTH-9328:03
suggestion[]=AUTH-9328|Default umask in /etc/login.defs could be more strict like 027|-|-|
auth_failed_logins_logged=0
ldap_auth_enabled=0
ldap_pam_enabled=0
password_min_days=-1
password_max_days=90
available_shell[]=/usr/bin/tcsh
available_shell[]=/usr/bin/csh
available_shell[]=/bin/tcsh
available_shell[]=/bin/csh
available_shell[]=/bin/bash
available_shell[]=/usr/bin/bash
session_timeout_enabled=0
suggestion[]=FILE-6310|To decrease the impact of a full /home file system, place /home on a separated partition|-|-|
suggestion[]=FILE-6310|To decrease the impact of a full /var file system, place /var on a separated partition|-|-|
file_systems_ext[]=none
suggestion[]=STRG-1840|Disable drivers like USB storage when not used, to prevent unauthorized storage or data theft|-|-|
suggestion[]=STRG-1846|Disable drivers like firewire storage when not used, to prevent unauthorized storage or data theft|-|-|
resolv_conf_option[]=timeout:15
resolv_conf_option[]=attempts:5
suggestion[]=NAME-4028|Check DNS configuration for the dns domain name|-|-|
suggestion[]=NAME-4404|Add the IP name and FQDN to /etc/hosts for proper name resolving|-|-|
localhost-mapped-to=::1
name_cache_used=0
package_manager[]=rpm
installed_packages=48
warning[]=PKGS-7383|YUM is not properly configured or registered for this platform (no repolist found)|-|-|
suggestion[]=PKGS-7384|Install package 'yum-utils' for better consistency checking of the package database|-|-|
suggestion[]=PKGS-7386|Install package yum-plugin-security if possible, to maintain security updates easier (yum install yum-plugin-security)|-|-|
software_repository_unsigned[]=photon
software_repository_unsigned[]=photon-updates
suggestion[]=PKGS-7398|Install a package audit tool to determine vulnerable packages|-|-|
installed_packages_array=|attr,2.4.47-3.ph1.x86_64,|bash,4.3.30-8.ph1.x86_64,|bzip2,1.0.6-6.ph1.x86_64,|ca-certificates,20160109-5.ph1.x86_64,|coreutils,8.25-2.ph1.x86_64,|curl,7.51.0-2.ph1.x86_64,|elfutils-libelf,0.165-2.ph1.x86_64,|expat,2.2.0-1.ph1.x86_64,|file,5.24-2.ph1.x86_64,|filesystem,1.0-10.ph1.x86_64,|gawk,4.1.3-2.ph1.x86_64,|gdbm,1.11-2.ph1.x86_64,|git,2.8.1-5.ph1.x86_64,|glib,2.47.6-2.ph1.x86_64,|glibc,2.22-9.ph1.x86_64,|gmp,6.0.0a-3.ph1.x86_64,|libcap,2.25-2.ph1.x86_64,|libffi,3.2.1-2.ph1.x86_64,|libgcc,5.3.0-3.ph1.x86_64,|libssh2,1.8.0-1.ph1.x86_64,|libstdc++,5.3.0-3.ph1.x86_64,|libxml2,2.9.4-3.ph1.x86_64,|lua,5.3.2-2.ph1.x86_64,|mpfr,3.1.3-2.ph1.x86_64,|ncurses,6.0-2.ph1.x86_64,|nspr,4.12-2.ph1.x86_64,|nss,3.25-1.ph1.x86_64,|openssl,1.0.2k-1.ph1.x86_64,|pcre,8.39-1.ph1.x86_64,|perl,5.22.1-4.ph1.x86_64,|perl-CGI,4.26-2.ph1.noarch,|perl-DBI,1.634-2.ph1.x86_64,|perl-YAML,1.15-2.ph1.noarch,|pkg-config,0.28-2.ph1.x86_64,|popt,1.16-2.ph1.x86_64,|pycurl,7.21.5-3.ph1.x86_64,|python-xml,2.7.13-1.ph1.x86_64,|python2,2.7.13-1.ph1.x86_64,|python2-libs,2.7.13-1.ph1.x86_64,|readline,6.3-4.ph1.x86_64,|rpm,4.11.2-11.ph1.x86_64,|rpm-devel,4.11.2-11.ph1.x86_64,|sed,4.2.2-2.ph1.x86_64,|sqlite-autoconf,3.11.0-2.ph1.x86_64,|urlgrabber,3.10.1-2.ph1.noarch,|yum,3.4.3-6.ph1.noarch,|yum-metadata-parser,1.1.4-2.ph1.x86_64,|zlib,1.2.8-3.ph1.x86_64,
package_audit_tool=
package_audit_tool_found=0
vulnerable_packages_found=0
ipv6_mode=auto
ipv6_only=0
nameserver[]=10.162.204.1
nameserver[]=10.166.1.1
default_gateway[]=10.192.255.253
network_interface[]=lo
network_interface[]=client
network_interface[]=bridge
network_mac_address[]=00:50:56:97:28:63
network_mac_address[]=00:50:56:97:39:f5
network_ipv4_address[]=172.16.0.1
network_ipv4_address[]=10.192.232.193
network_ipv4_address[]=127.0.0.1
network_ipv6_address[]=fe80::250:56ff:fe97:39f5/64
network_ipv6_address[]=fd01:1:2:2917:250:56ff:fe97:39f5/64
network_ipv6_address[]=fe80::250:56ff:fe97:2863/64
network_ipv6_address[]=fd01:1:2:2917:250:56ff:fe97:2863/64
network_ipv6_address[]=::1/128
network_listen_port[]=:::53|udp6|port-layer-serv|
network_listen_port[]=10.192.232.193:2376|tcp|docker-engine-s|
network_listen_port[]=127.0.0.1:2377|tcp|port-layer-serv|
network_listen_port[]=10.192.232.193:2377|tcp|port-layer-serv|
network_listen_port[]=10.192.232.193:2378|tcp|vicadmin|
network_listen_port[]=127.0.0.1:2379|tcp|port-layer-serv|
network_listen_port[]=127.0.0.1:6060|tcp|vic-init|
network_listen_port[]=127.0.0.1:6061|tcp|vicadmin|
network_listen_port[]=127.0.0.1:6062|tcp|docker-engine-s|
network_listen_port[]=127.0.0.1:6063|tcp|port-layer-serv|
network_listen_port[]=0.0.0.0:22|tcp|sshd|
network_listen_port[]=:::53|tcp6|port-layer-serv|
network_listen_port[]=:::22|tcp6|sshd|
suggestion[]=NETW-3032|Consider running ARP monitoring software (arpwatch,arpon)|-|-|
dhcp_client_running=0
arpwatch_running=0
printing_daemon=
imap_daemon=
pop3_daemon=
smtp_daemon=
firewall_software[]=iptables
firewall_no_logging[]=iptables
manual[]=Verify if there is a formal process for testing and applying firewall rules
manual[]=Verify all traffic is filtered the right way between the different security zones
manual[]=Verify if a list is available with all required services
manual[]=Make sure an explicit deny all is the default policy for all unmatched traffic
firewall_active=1
firewall_empty_ruleset=0
firewall_installed=1
firewall_software=iptables
suggestion[]=SSH-7408|Consider hardening SSH configuration|AllowTcpForwarding (YES --> NO)|-|
details[]=SSH-7408|sshd|desc:sshd option AllowTcpForwarding;field:AllowTcpForwarding;prefval:NO;value:YES;|
suggestion[]=SSH-7408|Consider hardening SSH configuration|ClientAliveCountMax (3 --> 2)|-|
details[]=SSH-7408|sshd|desc:sshd option ClientAliveCountMax;field:ClientAliveCountMax;prefval:2;value:3;|
suggestion[]=SSH-7408|Consider hardening SSH configuration|Compression (YES --> NO)|-|
details[]=SSH-7408|sshd|desc:sshd option Compression;field:Compression;prefval:NO;value:YES;|
suggestion[]=SSH-7408|Consider hardening SSH configuration|LogLevel (INFO --> VERBOSE)|-|
details[]=SSH-7408|sshd|desc:sshd option LogLevel;field:LogLevel;prefval:VERBOSE;value:INFO;|
suggestion[]=SSH-7408|Consider hardening SSH configuration|MaxAuthTries (6 --> 2)|-|
details[]=SSH-7408|sshd|desc:sshd option MaxAuthTries;field:MaxAuthTries;prefval:2;value:6;|
suggestion[]=SSH-7408|Consider hardening SSH configuration|MaxSessions (10 --> 2)|-|
details[]=SSH-7408|sshd|desc:sshd option MaxSessions;field:MaxSessions;prefval:2;value:10;|
suggestion[]=SSH-7408|Consider hardening SSH configuration|PermitRootLogin (YES --> NO)|-|
details[]=SSH-7408|sshd|desc:sshd option PermitRootLogin;field:PermitRootLogin;prefval:NO;value:YES;|
suggestion[]=SSH-7408|Consider hardening SSH configuration|Port (22 --> )|-|
details[]=SSH-7408|sshd|desc:sshd option Port;field:Port;prefval:;value:22;|
suggestion[]=SSH-7408|Consider hardening SSH configuration|TCPKeepAlive (YES --> NO)|-|
details[]=SSH-7408|sshd|desc:sshd option TCPKeepAlive;field:TCPKeepAlive;prefval:NO;value:YES;|
suggestion[]=SSH-7408|Consider hardening SSH configuration|AllowAgentForwarding (YES --> NO)|-|
details[]=SSH-7408|sshd|desc:sshd option AllowAgentForwarding;field:AllowAgentForwarding;prefval:NO;value:YES;|
ssh_daemon_running=1
syslog_daemon_present=1
syslog_daemon[]=systemd-journal
log_directory[]=/var/log
log_directory[]=/var/log
log_rotation_config_found=1
log_rotation_tool=logrotate
suggestion[]=BANN-7126|Add a legal banner to /etc/issue, to warn unauthorized users|-|-|
weak_banner_file[]=/etc/issue
suggestion[]=BANN-7130|Add legal banner to /etc/issue.net, to warn unauthorized users|-|-|
suggestion[]=ACCT-9622|Enable process accounting|-|-|
suggestion[]=ACCT-9626|Enable sysstat to collect accounting (no results)|-|-|
suggestion[]=ACCT-9628|Enable auditd to collect audit information|-|-|
linux_auditd_running=0
audit_daemon_running=0
tz_variable_empty=1
ntp_config_found=0
ntp_config_type_daemon=0
ntp_config_type_eventbased=0
ntp_config_type_scheduled=0
ntp_config_type_startup=0
ntp_daemon=
ntp_daemon_running=0
certificates=0
framework_grsecurity=0
framework_selinux=0
suggestion[]=FINT-4350|Install a file integrity tool to monitor changes to critical and sensitive files|-|-|
suggestion[]=TOOL-5002|Determine if automation tools are present for system management|-|-|
automation_tool_present=0
malware_scanner_installed=0
home_directory[]=/
home_directory[]=/home/vicadmin
home_directory[]=/root
home_directory[]=/var/lib/sshd
details[]=KRNL-6000|sysctl|desc:Disable IP source routing;field:net.ipv4.conf.all.forwarding;prefval:0;value:1;|
details[]=KRNL-6000|sysctl|desc:Disable/Ignore ICMP routing redirects;field:net.ipv6.conf.all.accept_redirects;prefval:0;value:1;|
details[]=KRNL-6000|sysctl|desc:Disable/Ignore ICMP routing redirects;field:net.ipv6.conf.default.accept_redirects;prefval:0;value:1;|
suggestion[]=KRNL-6000|One or more sysctl values differ from the scan profile and could be tweaked|-|-|
suggestion[]=HRDN-7230|Harden the system by installing at least one malware scanner, to perform periodic file system scans|-|Install a tool like rkhunter, chkrootkit, OSSEC|
compiler_installed=0
lynis_tests_done=194
report_datetime_end=2017-04-11 01:21:42
hardening_index=74
tests_executed=HRDN-7230|HRDN-7222|HRDN-7220|KRNL-6000|HOME-9350|HOME-9310|HOME-9302|FILE-7524|MALW-3284|MALW-3282|MALW-3280|MALW-3278|MALW-3276|MALW-3275|TOOL-5190|TOOL-5102|TOOL-5002|FINT-4350|FINT-4338|FINT-4330|FINT-4328|FINT-4326|FINT-4322|FINT-4318|FINT-4314|FINT-4310|MACF-6290|RBAC-6272|MACF-6232|MACF-6204|CONT-8102|CRYP-7902|TIME-3170|TIME-3148|TIME-3104|ACCT-9636|ACCT-9628|ACCT-9626|ACCT-9622|SCHD-7718|SCHD-7704|SCHD-7702|BANN-7130|BANN-7128|BANN-7126|BANN-7124|INSE-8002|LOGG-2180|LOGG-2170|LOGG-2154|LOGG-2150|LOGG-2148|LOGG-2146|LOGG-2142|LOGG-2138|LOGG-2240|LOGG-2230|LOGG-2210|LOGG-2136|LOGG-2132|LOGG-2130|SQD-3602|PHP-2211|LDAP-2219|DBS-1880|DBS-1860|DBS-1840|DBS-1826|DBS-1820|DBS-1818|DBS-1804|SNMP-3302|SSH-7440|SSH-7408|SSH-7404|SSH-7402|HTTP-6702|HTTP-6622|FIRE-4590|FIRE-4586|FIRE-4524|FIRE-4513|FIRE-4512|FIRE-4508|FIRE-4502|MAIL-8880|MAIL-8860|MAIL-8838|MAIL-8820|MAIL-8814|MAIL-8802|PRNT-2314|PRNT-2304|NETW-3032|NETW-3030|NETW-3028|NETW-3015|NETW-3012|NETW-3008|NETW-3006|NETW-3004|NETW-3001|NETW-2705|NETW-2704|NETW-2600|PKGS-7410|PKGS-7398|PKGS-7387|PKGS-7386|PKGS-7384|PKGS-7383|PKGS-7308|NAME-4408|NAME-4406|NAME-4404|NAME-4402|NAME-4304|NAME-4230|NAME-4202|NAME-4034|NAME-4032|NAME-4028|NAME-4020|NAME-4018|NAME-4016|STRG-1920|STRG-1846|STRG-1842|STRG-1840|FILE-6430|FILE-6376|FILE-6374|FILE-6372|FILE-6368|FILE-6362|FILE-6354|FILE-6344|FILE-6336|FILE-6332|FILE-6329|FILE-6323|FILE-6310|SHLL-6290|SHLL-6230|SHLL-6220|SHLL-6211|AUTH-9408|AUTH-9402|AUTH-9328|AUTH-9308|AUTH-9288|AUTH-9286|AUTH-9283|AUTH-9282|AUTH-9278|AUTH-9268|AUTH-9266|AUTH-9264|AUTH-9262|AUTH-9252|AUTH-9250|AUTH-9242|AUTH-9240|AUTH-9234|AUTH-9228|AUTH-9226|AUTH-9222|AUTH-9216|AUTH-9208|AUTH-9204|PROC-3614|PROC-3612|PROC-3602|KRNL-5830|KRNL-5820|KRNL-5730|KRNL-5728|KRNL-5726|KRNL-5723|KRNL-5695|KRNL-5677|KRNL-5622|BOOT-5260|BOOT-5202|BOOT-5184|BOOT-5177|BOOT-5155|BOOT-5142|BOOT-5139|BOOT-5121|BOOT-5116|BOOT-5108|BOOT-5104|CORE-1000|
tests_skipped=MALW-3288|MALW-3286|TOOL-5104|FINT-4402|FINT-4336|FINT-4334|FINT-4315|MACF-6234|MACF-6208|CONT-8108|CONT-8107|CONT-8106|CONT-8104|CONT-8004|TIME-3160|TIME-3136|TIME-3132|TIME-3128|TIME-3124|TIME-3120|TIME-3116|TIME-3112|TIME-3106|ACCT-9662|ACCT-9660|ACCT-9656|ACCT-9654|ACCT-9652|ACCT-9650|ACCT-9634|ACCT-9632|ACCT-9630|ACCT-2760|ACCT-2754|SCHD-7724|SCHD-7720|BANN-7113|INSE-8050|INSE-8016|INSE-8006|INSE-8004|LOGG-2192|LOGG-2190|LOGG-2164|LOGG-2162|LOGG-2160|LOGG-2152|LOGG-2134|SQD-3680|SQD-3630|SQD-3624|SQD-3620|SQD-3616|SQD-3614|SQD-3613|SQD-3610|SQD-3606|SQD-3604|PHP-2378|PHP-2376|PHP-2374|PHP-2372|PHP-2368|PHP-2320|LDAP-2224|DBS-1888|DBS-1886|DBS-1884|DBS-1882|DBS-1816|SNMP-3306|SNMP-3304|HTTP-6720|HTTP-6716|HTTP-6714|HTTP-6712|HTTP-6710|HTTP-6708|HTTP-6706|HTTP-6704|HTTP-6643|HTTP-6641|HTTP-6640|HTTP-6632|HTTP-6626|HTTP-6624|FIRE-4540|FIRE-4538|FIRE-4536|FIRE-4534|FIRE-4532|FIRE-4530|FIRE-4526|FIRE-4520|FIRE-4518|MAIL-8920|MAIL-8818|MAIL-8817|MAIL-8816|PRNT-2420|PRNT-2418|PRNT-2316|PRNT-2308|PRNT-2307|PRNT-2306|PRNT-2302|NETW-3014|PKGS-7394|PKGS-7393|PKGS-7392|PKGS-7390|PKGS-7388|PKGS-7382|PKGS-7381|PKGS-7380|PKGS-7378|PKGS-7370|PKGS-7366|PKGS-7354|PKGS-7352|PKGS-7350|PKGS-7348|PKGS-7346|PKGS-7345|PKGS-7330|PKGS-7328|PKGS-7322|PKGS-7320|PKGS-7314|PKGS-7312|PKGS-7310|PKGS-7306|PKGS-7304|PKGS-7303|PKGS-7302|PKGS-7301|NAME-4306|NAME-4238|NAME-4236|NAME-4232|NAME-4210|NAME-4206|NAME-4204|NAME-4036|NAME-4026|NAME-4024|STRG-1930|STRG-1928|STRG-1926|STRG-1906|STRG-1904|STRG-1902|FILE-6410|FILE-6439|FILE-6330|FILE-6312|FILE-6311|SHLL-6202|AUTH-9406|AUTH-9340|AUTH-9306|AUTH-9304|AUTH-9254|AUTH-9489|AUTH-9218|AUTH-9212|PROC-3604|KRNL-5788|KRNL-5770|KRNL-5831|KRNL-5745|BOOT-5180|BOOT-5165|BOOT-5159|BOOT-5126|BOOT-5261|BOOT-5124|BOOT-5122|BOOT-5106|BOOT-5102|
finish=true

**Comment by dougm** [comment]
Thanks @maplain , we can follow up with a separate story to take action on the suggestions.

**Comment by maplain** [comment]
sure. let me know what I can help here. @dougm

File: tests/manual-test-cases/Group20-Security/20-1-Appliance-Audit.robot [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    Should Be Equal As Integers  ${rc}  0
+
+Lynis Audit System
+    [Arguments]  ${target}=%{VCH-IP}  ${user}=root  ${password}=%{TEST_PASSWORD}  ${lynis_path}=/usr/local/lynis
+    Log To Console  \nBegin auditing appliance...
+    Open Connection  ${target}
+    Login  ${user}  ${password}
+    ${rc}=  Execute Command  cd ${lynis_path} && ./lynis audit system  return_stdout=False  return_rc=True
+    Should Be Equal As Integers  ${rc}  0
+    ${rc}=  Execute Command  cp /var/log/lynis* .  return_stdout=False  return_rc=True
```
Comment (hmahmood) [inline tests/manual-test-cases/Group20-Security/20-1-Appliance-Audit.robot]: This copies the results to root's home directory on the VCH, right? Should we copy it back down to the client running the test, as robot framework does the same with its reports?

File: tests/resources/VCH-Util.robot [hunk 2]
```diff
@@ -393,3 +393,10 @@ Rollback
     ${rc}  ${output}=  Run And Return Rc And Output  bin/vic-machine-linux upgrade --debug 1 --name=%{VCH-NAME} --target=%{TEST_URL} --user=%{TEST_USERNAME} --password=%{TEST_PASSWORD} --force=true --compute-resource=%{TEST_RESOURCE} --timeout %{TEST_TIMEOUT} --rollback
     Should Contain  ${output}  Completed successfully
     Should Be Equal As Integers  ${rc}  0
+
+Enable VCH SSH
+    [Arguments]  ${vic-machine}=bin/vic-machine-linux  ${rootpw}=%{TEST_PASSWORD}  ${target}=%{TEST_URL}  ${password}=%{TEST_PASSWORD}  ${thumbprint}=%{TEST_THUMBPRINT}  ${name}=%{VCH-NAME}  ${user}=%{TEST_USERNAME}  ${resource}=%{TEST_RESOURCE}
+    Log To Console  \nEnalbe SSH on vch...
```
Comment (mhagen-vmware) [inline tests/resources/VCH-Util.robot]: enable

File: tests/manual-test-cases/Group20-Security/20-1-Appliance-Audit.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+Test 20-1 - Appliance OS Security Audit
+======
+
+#Purpose:
+To detect known OS security issues and warnings
+
+#References:
+
```
Comment (mhagen-vmware) [inline tests/manual-test-cases/Group20-Security/20-1-Appliance-Audit.md]: Can you add at least one reference to something related to how lynis works?

File: tests/manual-test-cases/Group20-Security/20-1-Appliance-Audit.robot [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+*** Settings ***
+Documentation  Test 20-01 - Appliance-Audit
+Suite Setup  Setup Test Environment
+Suite Teardown  Clean up VIC Appliance And Local Binary
+Resource  ../../resources/Util.robot
+
+*** Keywords ***
+Setup Test Environment
+    [Arguments]  ${version}=
```
Comment (mhagen-vmware) [inline tests/manual-test-cases/Group20-Security/20-1-Appliance-Audit.robot]: Can you make the default at least ${EMPTY} to be explicit, I am not really sure what the behavior of leaving it like this.

File: tests/manual-test-cases/Group20-Security/20-1-Appliance-Audit.robot [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    Run Keyword If  '${version}' == '${EMPTY}'  Install VIC Appliance To Test Server
+    Run Keyword Unless  '${version}' == '${EMPTY}'  Install VIC with version to Test Server  ${version}
+    Run Keyword If  '${version}' == '${EMPTY}'  Enable VCH SSH
+    Run Keyword Unless  '${version}' == '${EMPTY}'  Enable VCH SSH  vic/vic-machine-linux
+
+Provision lynis
+    [Arguments]  ${target}=%{VCH-IP}  ${user}=root  ${password}=%{TEST_PASSWORD}
```
Comment (mhagen-vmware) [inline tests/manual-test-cases/Group20-Security/20-1-Appliance-Audit.robot]: root -> %{TEST_USERNAME}?

File: tests/manual-test-cases/Group20-Security/20-1-Appliance-Audit.robot [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    Log To Console  \nProvision lynis to vch appliance...
```
Comment (mhagen-vmware) [inline tests/manual-test-cases/Group20-Security/20-1-Appliance-Audit.robot]: nice use of \n, I wish we did this more

File: tests/manual-test-cases/Group20-Security/20-1-Appliance-Audit.robot [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    Open Connection  ${target}
+    Login  ${user}  ${password}
+    ${rc}=  Execute Command  rpm --rebuilddb  return_stdout=False  return_rc=True
```
Comment (mhagen-vmware) [inline tests/manual-test-cases/Group20-Security/20-1-Appliance-Audit.robot]: for debuggability of this test, should we really be throwing away stdout here and below?

File: tests/manual-test-cases/Group20-Security/20-1-Appliance-Audit.robot [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    Should Be Equal As Integers  ${rc}  0
+    ${rc}=  Execute Command  tdnf install -y yum  return_stdout=False  return_rc=True
+    Should Be Equal As Integers  ${rc}  0
+    ${rc}=  Execute Command  yum install -y awk sed git  return_stdout=False  return_rc=True
+    Should Be Equal As Integers  ${rc}  0
+    ${rc}=  Execute Command  cd /usr/local && git clone https://github.com/CISOfy/lynis  return_stdout=False  return_rc=True
+    Should Be Equal As Integers  ${rc}  0
+
+Lynis Audit System
+    [Arguments]  ${target}=%{VCH-IP}  ${user}=root  ${password}=%{TEST_PASSWORD}  ${lynis_path}=/usr/local/lynis
```
Comment (mhagen-vmware) [inline tests/manual-test-cases/Group20-Security/20-1-Appliance-Audit.robot]: root -> %{TEST_USERNAME}

File: tests/manual-test-cases/Group20-Security/20-1-Appliance-Audit.md (hunk 3 above)
Comment (maplain) [inline tests/manual-test-cases/Group20-Security/20-1-Appliance-Audit.md]: sure

File: tests/manual-test-cases/Group20-Security/20-1-Appliance-Audit.robot (hunk 4 above)
Comment (maplain) [inline tests/manual-test-cases/Group20-Security/20-1-Appliance-Audit.robot]: sure

File: tests/manual-test-cases/Group20-Security/20-1-Appliance-Audit.robot (hunk 5 above)
Comment (maplain) [inline tests/manual-test-cases/Group20-Security/20-1-Appliance-Audit.robot]: it's the root username for vch appliance, shouldn't be the default %{TEST_USERNAME}

**Pull request merged**